pkgs/top-level/all-packages.nix

```
@@ -36653,18 +36653,18 @@ with pkgs;
 
   lie = callPackage ../applications/science/math/LiE { };
 
-  magma = callPackage ../development/libraries/science/math/magma {
+  inherit (callPackage ../development/libraries/science/math/magma {
     inherit (llvmPackages_rocm) openmp;
-  };
+  }) magma magma_2_7_1 magma_2_6_2;
 
   magma-cuda = magma.override {
-    useCUDA = true;
-    useROCM = false;
+    cudaSupport = true;
+    rocmSupport = false;
   };
 
-  magma-hip = magma.override {
-    useCUDA = false;
-    useROCM = true;
+  magma-hip = magma_2_6_2.override {
```

```suggestion
  magma-hip = magma.override {
```
why not just use magma here?

---

AMD won't work with anything newer than 2.6.2 right now so we need to have them stay on that :(
(this PR changes the default to 2.7.1)

---

ahhh ok thanks for explaining! let's add a comment explaining that in the code then